src/main/scala/org/apache/spark/sql/delta/sources/DeltaDataSource.scala

```diff
@@ -330,4 +331,56 @@ object DeltaDataSource extends DatabricksLogging {
       None
     }
   }
+
+  /**
+   * Extends the DataFrameReader API by adding a delta function
+   * Usage:
+   * {{{
+   * spark.read.delta(path)
+   * }}}
+   */
+  implicit class DeltaDataFrameReader(val reader: DataFrameReader) extends AnyVal {
```

Comment:
DeltaDataSource is a private API. Could you create a new package file under `io.delta` and move these codes there? You can take a look at https://github.com/databricks/spark-avro/blob/branch-4.0/src/main/scala/com/databricks/spark/avro/package.scala as an example.

Reply:
In addition,  the class name look not consistent. Can we use `DeltaDataFrameReader`, `DeltaDataFrameWriter`, `DeltaDataStreamReader`, `DeltaDataStreamWriter`?

Reply:
Great Idea 💯  I have taken into account your feedback and have created the ```io.delta.sources``` package.
Let me know if you want me to change the name.
I have also renamed the different implicit classes as requested.